jbpm-designer-backend/src/main/java/org/jbpm/designer/bpmn2/impl/Bpmn2JsonMarshaller.java

```diff
@@ -1897,15 +1906,13 @@ protected void marshallTask(Task task,
             taskType = "Send";
             SendTask st = (SendTask) task;
             if (st.getMessageRef() != null) {
-                properties.put("messageref",
-                               st.getMessageRef().getId());
+                setMessageRefProperties(properties, st.getMessageRef());
             }
         } else if (task instanceof ReceiveTask) {
             taskType = "Receive";
             ReceiveTask rt = (ReceiveTask) task;
             if (rt.getMessageRef() != null) {
```

Comment:
We don't need this `if`, as it is done inside `setMessageRefProperties` implementation.